### DISCUSSION TITLE
Can the filtering speed be improved?

### DISCUSSION THREAD

**Discussion by DevAnyKR** [moneymanagerex/moneymanagerex#5906] [Show and tell]
![image](https://github.com/moneymanagerex/moneymanagerex/assets/110871727/7f63b644-8113-4525-ae75-a8f86db1650a)

problem: The moment I type in a search term, the program freezes. (7500 records)
The UI responds after a few seconds, but the search results are unknown.

So, I created a search-only program. Search in real time.

just sample, haha...

![image](https://github.com/moneymanagerex/moneymanagerex/assets/110871727/324964e3-ebc7-495e-9084-c37e8c93e2f8)

https://github.com/DevelopmentAnything/MMDX/releases/

**Comment by vomikan** [comment]
I can't reproduce it.

**Reply by DevAnyKR** [reply]
![스크린샷 2023-05-12 212413](https://github.com/moneymanagerex/moneymanagerex/assets/110871727/a30015df-e291-4e7d-9940-15e2154a4289)

1. input search string
2. Enter key press
3. Freezing...

**Reply by vomikan** [reply]
I don't have a reproducible problem with the search, but I found a similar bug.

1) Enter some search string into the search fiels
2) Click on any transaction
3) Press Ctrl+A
4) Set a focus into the Search feld
5) Press Enter

https://github.com/moneymanagerex/moneymanagerex/assets/6836805/58a8529f-f6da-4740-b8b5-6ac15a69704a

**Reply by DevAnyKR** [reply]
It counts for a moment and then freezes. 
My records are over 7500. Less than 1,000 records are cute.
Stop at 7411. -100 counts only...
There is no recovery and the process must be forcefully terminated.
If I use Ctrl+A , the program freezes forever.

search field: CAR
Ctrl+A
focused search field and Enter key...
![스크린샷 2023-05-15 224016](https://github.com/moneymanagerex/moneymanagerex/assets/110871727/00b7567f-defb-4743-bb69-234f2268a118)

Task manager - terminate Process
restart MMEX
![스크린샷 2023-05-15 224932](https://github.com/moneymanagerex/moneymanagerex/assets/110871727/3ee1052d-37c6-4d89-aed1-a14cbc5b16a5)

There is no problem with database and storage.
Because other programs(my project) process MMEX DB very quickly.

Is it related to Korean? Otherwise, Recommend testing on tens of thousands of records.

**Reply by n-stein** [reply]
Searching my ~17000 record database in the All Transactions view takes about 15 seconds.

I think you may be underestimating what the search field is doing here. It is searching through every field of every transaction including the descriptions of all attachments of every transaction. I'm not sure how much improvement can be made and still retain that functionality.

Of course this is affected by how many transactions are in the view. If you can limit to a specific account or time frame it is much faster.

**Reply by DevAnyKR** [reply]
When I tried again today, the unresponsive disappeared, but the UI freezes for several seconds.
Another question.
Are all transactions in All Transactions Grid loaded into memory? Actual memory usage is 23 MB.
Searches are quite slow. Only in MMEX...

Search conditions cannot be limited by account or period. Because I don't know where the search target is.
But I need to be able to target the search field.

Example of filtering by field
![image](https://github.com/moneymanagerex/moneymanagerex/assets/110871727/2563aaa0-01c2-4228-96c2-5df0945a3417)

**Reply by vomikan** [reply]
https://github.com/moneymanagerex/moneymanagerex/issues/5915
